contrib/guix/guix-build.sh

```diff
@@ -314,7 +314,7 @@ EOF
                                         ${BASE_CACHE:+BASE_CACHE="$BASE_CACHE"} \
                                         DISTSRC="$(DISTSRC_BASE=/distsrc-base && distsrc_for_host "$HOST")" \
                                         OUTDIR=/outdir \
-                                      bash -c "cd /bitcoin && bash contrib/guix/libexec/build.sh"
+                                      bash -c "cd /bitcoin && umask 0022 && bash contrib/guix/libexec/build.sh"
```

Comment:
What is the rationale for not moving this *into* `contrib/guix/libexec/build.sh`?

Reply:
> What is the rationale for not moving this _into_ `contrib/guix/libexec/build.sh`?

None at all! Will move it in.

> (also, is this a temporary workaround until Guix isolates their builds better, or are we stuck with it, in the first case let's add a comment)

I think I may have good standing to at least have `guix environment --container` set the `umask` value, so I will add a comment in case that patch gets approved.